src/main/java/org/cactoos/text/TextEnvelope.java

```diff
@@ -73,8 +73,23 @@ public final int hashCode() {
         return new UncheckedScalar<>(this.origin).value().hashCode();
     }
 
+    // @todo #942:30min Refactor TextEnvelope.equals(). Current implementation
+    // of TextEnvelope.equals() uses some things that we must avoid, like more
+    // than one return on method, instance of usage and typecasting. Refactor
+    // this method so we can get rid of these smelly things.
+    //
     @Override
+    @SuppressWarnings("PMD.OnlyOneReturn")
     public final boolean equals(final Object obj) {
-        return new UncheckedScalar<>(this.origin).value().equals(obj);
+        if (this == obj) {
+            return true;
+        }
+        if (!(obj instanceof TextEnvelope)) {
```

Comment:
@paulodamaso let's make program against the `Text` interface, not the implementation